### ISSUE TITLE
Player positions

### ISSUE THREAD

**Issue by youbloodywombat** [yasp-dota/yasp#1171]
Hi,

Are there any plans to improve location data, by getting times as well as x,y co-ordinates from replaying parsing? I think there could be some cool stuff to analysis around team rotations particularly with supports in the first 20 mins.

>  "posData": {
>                 "sen": [], 
>                 "obs": [], 
>                 "lane_pos": [
>                     {
>                         "y": 61, 
>                         "x": 10, 
>                         "value": 1
>                     }, 
>                     {
>                         "y": 65, 
>                         "x": 12, 
>                         "value": 1
>                     }, 
>                     ....

Thanks
Jeff

**Comment by howardchung** [comment]
The parser does extract this data, but we don't persist it since it takes up a ton of space.  If it's something that can be computed and stored more simply, we can do that.

**Comment by youbloodywombat** [comment]
The x & y co-ordinates could be identified into rough areas i.e. Mid Lane Top, river rune spot. Then limit what is stored to keep the first time someone visited an area.

**Comment by howardchung** [comment]
We do have a mapping of each x/y pair to one of five buckets (mid/top/bot/rjungle/djungle).  If you give an example of what kind of data you're looking for we can come up with a plan

**Comment by youbloodywombat** [comment]
It'd look something like:
{ area: mid, time: 10}
,{ area: bot, time 150}
,{ area: mid, time 250}

**Comment by howardchung** [comment]
Yeah, I think this could be done.  I'm not sure when I'll have time for it, so putting in backlog, but if someone wants to tackle it I'm happy to help them get started.

**Comment by DarrenGuinness** [comment]
Could this be useful? https://www.npmjs.com/package/density-clustering (DBSCAN). I've used it for similar clustering. Just gotta tune the epsilon and minpts parameters.

**Comment by howardchung** [comment]
Maybe, I haven't looked at using a library since the current task is pretty simple.
